libretro-common/include/streams/chd_stream.h

```diff
@@ -57,6 +57,8 @@ int64_t chdstream_seek(chdstream_t *stream, int64_t offset, int whence);
 
 ssize_t chdstream_get_size(chdstream_t *stream);
 
+uint32_t chdstream_get_pregap(chdstream_t* stream);
+
```

Comment:
In my testing, the `pgtype` was always "`VMODE1_RAW`", which didn't match the `type` "`MODE1_RAW`", so the pregap was never acknowledged:

https://github.com/libretro/RetroArch/blob/45deb2c2182ff6309f4dbffb08d61aff070455e9/libretro-common/streams/chd_stream.c#L248-L252

It looks like the MAME code is expecting the "V" and just processes the remaining portion of the string as a separate format that may differ from the data content: https://github.com/mamedev/mame/blob/13247311f650d2190ca50f05f2879c63a3ad0b6b/src/lib/util/cdrom.cpp#L1225-L1228

I briefly looked at modifying the logic to ignore the leading `V` (similar to MAME) and ran into another issue farther down the line:

`track_start` is calculated using the above `pregap` value: https://github.com/libretro/RetroArch/blob/45deb2c2182ff6309f4dbffb08d61aff070455e9/libretro-common/streams/chd_stream.c#L257

And only used to simulate the pregap when reading: https://github.com/libretro/RetroArch/blob/45deb2c2182ff6309f4dbffb08d61aff070455e9/libretro-common/streams/chd_stream.c#L337-L343

Since `pregap` was always 0 due to the leading `V`, attempting to read sector 1 always returned pregap data. With the change to calculate the correct `track_start`, `stream->offset` is still calculated from the start of the pregap, so asking for sector 1 would still return the same pregap sector. In the first case, the pregap data came directly out of the CHD. In the second case, it came from the virtualized logic highlighted above.

My solution was to implement a secondary function that returns the number of pregap sectors so I could offset the read. I'm not sure if this is the best solution, but it maintains the existing functionality for anything that might rely on it.